src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferFundsSection.tsx

```diff
@@ -153,7 +154,7 @@ const TransferFundsSection = ({
     [safeBalances, selectedTokenAddress],
   );
   const selectedSafe = safes.find(
-    (s) => s.contractAddress === safe?.profile.walletAddress,
+    (safe) => safe.contractAddress === inputtedSafe?.profile.walletAddress,
```

Comment:
You might want to consider extracting this into an util method similarly to `getSelectedSafeBalance` or `getSelectedNFTData`. Seems like this is needed in 2 other places so could save a few LOC.

Reply:
@jakub nice idea